### ISSUE TITLE
ContextualPrecision endless generation

### ISSUE THREAD

**Issue by enrico-stauss** [confident-ai/deepeval#1205]
**Describe the bug**
For longer/more complex contexts, my evaluator stalls in the `ContextualPrecision._generate_verdicts()` in an endless generation loop. I can reproduce this in an LLM playground but unfortunately I can't share the exact contexts here.

I'm 'only' using llama 3.1 70B as evaluator, so using a more capable model would probably help, but I was actually able to resolve this by simply adding the number of nodes to the prompt *(this has helped me before with similar issues)*. 

I'm happy to submit a PR containing the change, then you can decide whether to integrate or not. Are you interested?

**Comment by penguine-ip** [comment]
Hey @enrico-stauss I would be interested - but before you do anything do you know why it stalls in an endless loop?

**Comment by enrico-stauss** [comment]
Hi @penguine-ip , I believe it is the phenomenon that is also documented here: https://www.reddit.com/r/LocalLLaMA/comments/1ap8mxh/what_causes_llms_to_fall_into_repetitions_while/

I had observed similar issues before when I just provided a relatively long/complex prompt into the Groq playground - this time it happened on Azure Foundry. The framework I'm using for the LLM API is LlamaIndex by the way. The prompt itself is formatted through the `ContextualPrecisionTemplate.generate_verdicts` template and the contexts are 10 nodes of which some contain relatively complex html tables. 
I could probably demonstrate it to you if we would set up a meeting.

I think the reason that it enters an endless loop might be that in a long list of very long contexts the LLM fails to recognize how many sources actually are provided which is also why it can be fixed easily by just explicitly providing the true number of sources in the prompt.
